rpc/src/server/router.rs

```diff
@@ -14,66 +14,80 @@ pub(crate) fn create_routes(is_sandbox: bool) -> PathTree<Handler> {
+    routes.handle(Method::GET, "/chains/:chain_id/blocks/:block_id/protocols", shell_handler::get_block_protocols);
+    routes.handle(Method::GET, "/chains/:chain_id/blocks/:block_id/hash", shell_handler::get_block_hash);
+    routes.handle(Method::GET, "/chains/:chain_id/blocks/:block_id/operation_hashes", shell_handler::get_block_operation_hashes);
+    routes.handle(Method::GET, "/chains/:chain_id/blocks/:block_id/context/raw/bytes", shell_handler::context_raw_bytes);
+    routes.handle(Method::GET, "/chains/:chain_id/blocks/:block_id/context/raw/bytes/*any", shell_handler::context_raw_bytes);
+    routes.handle(Method::GET, "/chains/:chain_id/blocks/:block_id/metadata", shell_handler::chains_block_id_metadata);
+    routes.handle(Method::POST, "/injection/operation", shell_handler::inject_operation);
+    routes.handle(Method::GET, "/workers/prevalidators", shell_handler::worker_prevalidators);
     // TODO: TE-174: just for sandbox
     if is_sandbox {
-        routes.handle("/injection/block", shell_handler::inject_block);
+        routes.handle(Method::POST, "/injection/block", shell_handler::inject_block);
     }
 
     // Shell rpcs - routed through ffi calls
-    routes.handle("/chains/:chain_id/blocks/:block_id/helpers/preapply/operations", shell_handler::preapply_operations);
-    routes.handle("/chains/:chain_id/blocks/:block_id/helpers/preapply/block", shell_handler::preapply_block);
+    routes.handle(Method::GET, "/chains/:chain_id/blocks/:block_id/helpers/preapply/operations", shell_handler::preapply_operations);
+    routes.handle(Method::GET, "/chains/:chain_id/blocks/:block_id/helpers/preapply/block", shell_handler::preapply_block);
 
     // Protocol rpcs - implemented
-    routes.handle("/chains/:chain_id/blocks/:block_id/context/constants", protocol_handler::context_constants);
-    routes.handle("/chains/:chain_id/blocks/:block_id/helpers/baking_rights", protocol_handler::baking_rights);
-    routes.handle("/chains/:chain_id/blocks/:block_id/helpers/endorsing_rights", protocol_handler::endorsing_rights);
-    routes.handle("/chains/:chain_id/blocks/:block_id/votes/listings", protocol_handler::votes_listings);
+    routes.handle(Method::GET, "/chains/:chain_id/blocks/:block_id/context/constants", protocol_handler::context_constants);
+    routes.handle(Method::GET, "/chains/:chain_id/blocks/:block_id/helpers/baking_rights", protocol_handler::baking_rights);
+    routes.handle(Method::GET, "/chains/:chain_id/blocks/:block_id/helpers/endorsing_rights", protocol_handler::endorsing_rights);
+    routes.handle(Method::GET, "/chains/:chain_id/blocks/:block_id/votes/listings", protocol_handler::votes_listings);
 
     // Other Protocol rpcs - routed through ffi calls
-    routes.handle("/chains/:chain_id/blocks/:block_id/*any", protocol_handler::call_protocol_rpc);
+    // TODO: rework this, this should accept more methods
```

Comment:
@adonagy 
refactor this methods, to support multiple, at least for protocol we need to support them